src/client/Client.cc

```diff
@@ -4278,6 +4285,10 @@ void Client::add_update_cap(Inode *in, MetaSession *mds_session, uint64_t cap_id
 
   if (issued & ~old_caps)
     signal_cond_list(in->waitfor_caps);
+
+  // HACK: Just remove all caps in the client.
+  //       We force return true anyways
+  remove_all_caps(in);
```

Comment:
I couldn't early return from `add_update_cap` since it does some setup that seems required.
I just effectively remove all caps at the end of it inline though.